### PR TITLE
fix(agw): Fixed ENB Complete Reset Scenario

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Fixed ENB Complete Reset Scenario

## Summary
As per the Github issue https://github.com/magma/magma/issues/12583, the enb complete reset testcase was failing after the merge of PR #70. The testcase was getting stuck at the detach procedure waiting for detach accept indication and was failing with timeout. This PR fixes the issue with the enb complete reset scenario.

## Test plan
Verified with sanity and as individual test case

### CODE REVIEW

File: TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c [hunk 1]
```diff
@@ -818,6 +818,28 @@ PRIVATE S16 nbBldResetReq
...
+
+   if(resetMsgInfo->type != NB_COMPLETE_RESET && resetMsgInfo->type != NB_PARTIAL_RESET)
+   {
+      NB_LOG_ERROR(&nbCb, "Invalid Reset Type");
+      NB_FREE_EVNT(resetReqPdu);
+      RETVALUE(RFAILED);
+   }
+
+   // Allocate memory for storing UE Id pair to be reset
+   ret = cmAllocEvnt(sizeof(S1apPdu), NB_SZ_MEM_SDU_SIZE, &nbCb.mem,
```
Comment (pruthvihebbani) [inline TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c:830]: ret is not used anywhere. You can directly check the return type of the fun against ROK/RFAILED -> 
if ((cmAllocEvnt)!=ROK)

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

File: TestCntlrApp/src/enbApp/nb.c [hunk 2]
```diff
@@ -938,6 +938,26 @@ PUBLIC S16 NbEnbResetReqHdl
       while (cmHashListGetNext(&(nbCb.ueCbLst), (PTR)prev, (PTR *)&ueCb) == ROK)
       {
 	  if (ueCb->enbId ==  resetReq->u.completeRst.enbId) {
+             resetMsgInfo.s1apIdCnt++;
```
[the remark this answers is not in the document]
Comment (VinashakAnkitAman) [reply TestCntlrApp/src/enbApp/nb.c:941]: For getting the count of UEs associated to this eNB. Based on this count, resetMsgInfo.enbUeS1apIdLst and resetMsgInfo.mmeUeS1apIdLst will be allocated of that size and then in the next loop these lists will be filled with UE Ids.

[the remark this answers is not in the document]
Comment (pruthvihebbani) [reply TestCntlrApp/src/enbApp/nb.c:941]: Can you add a brief description here

**Review by pruthvihebbani — APPROVED** [review]
lgtm

File: TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c [hunk 3]
```diff
@@ -818,6 +813,32 @@ PRIVATE S16 nbBldResetReq
...
+      NB_LOG_ERROR(&nbCb, "Invalid Reset Type");
+      NB_FREE_EVNT(resetReqPdu);
+      RETVALUE(RFAILED);
+   }
+
+   // Allocate memory for storing UE Id pair to be reset
+   if(cmAllocEvnt(sizeof(S1apPdu), NB_SZ_MEM_SDU_SIZE, &nbCb.mem,
+         (Ptr *)&resetReqPdu1) != ROK)
+   {
+      NB_LOG_ERROR(&nbCb, "cmAllocEvnt failed");
```
Comment (rsarwad) [inline TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c:828]: On failing to create memory for resetReqPdu1, free memeory for resetReqPdu

File: TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c [hunk 4]
```diff
@@ -818,6 +813,32 @@ PRIVATE S16 nbBldResetReq
...
+
+   if(resetMsgInfo->type != NB_COMPLETE_RESET && resetMsgInfo->type != NB_PARTIAL_RESET)
+   {
+      NB_LOG_ERROR(&nbCb, "Invalid Reset Type");
+      NB_FREE_EVNT(resetReqPdu);
+      RETVALUE(RFAILED);
+   }
+
+   // Allocate memory for storing UE Id pair to be reset
+   if(cmAllocEvnt(sizeof(S1apPdu), NB_SZ_MEM_SDU_SIZE, &nbCb.mem,
```
Comment (rsarwad) [inline TestCntlrApp/src/enbApp/nb_s1ap_hndlr.c:825]: Check whether resetReqPdu1 is getting freed.

**Review by rsarwad — CHANGES_REQUESTED** [review]
Left few comments

**Review by rsarwad — APPROVED** [review]
LGTM

**Pull request merged**